Client/Models/CommunicationHandler.cs

```diff
@@ -9,14 +10,108 @@ public static void HandleMessage(Message message)
         {
             switch (message.MessageType)
             {
+                case MessageType.RegisterClientResponse:
+                    HandleRegisterClientResponse(message);
+                    break;
+                case MessageType.SignInClientResponse:
+                    HandleSignInClientResponse(message);
+                    break;
+                case MessageType.RegisterGroupResponse:
+                    HandleRegisterGroupResponse(message);
+                    break;
                 case MessageType.RequestGroupsResponse:
-                    List<Group> list = new List<Group>();
-                    // Convert Messenger_Client.Group to Shared.Group.
-                    message.GroupList.ForEach(group => list.Add(new Group(group)));
+                    HandleRequestGroupsResponse(message);
+                    break;
+                case MessageType.JoinGroupResponse:
+                    HandleJoinGroupResponse(message);
+                    break;
+                case MessageType.ChatMessage:
+                    HandleChatMessage(message);
+                    break;
+            }
+        }
+        
 
-                    // TODO: assign to some viewmodel list (with writelock?).
+        private static void HandleRegisterClientResponse(Message message)
+        {
+            switch (message.ClientId)
+            {
+                case -1:
+                    Console.WriteLine("Naam is al bezet!");
+                    break;
+                default:
+                    Console.WriteLine("Account aangemaakt!");
+                    Client.Instance.Id = message.ClientId;
```

Comment:
Naam is al bezet?, namen hoefden toch niet uniek te zijn?
Daarnaast is het misschien beter om //TODO te doen, op deze manier zijn zaken die nog gedaan moeten worden beter te herkennen en op te zoeken.

Reply:
ik heb overal waar ik denk dat nodig is een TODO: toegevoegd. 
Ook heb ik het veranderd van naam al bezet naar e-mail al bezet want het email adress hoort wel uniek te zijn